engine/src/position.rs

```diff
@@ -98,8 +102,10 @@ impl Position {
                 "/" => {
                     position = position.to(&Direction::SW);
                 }
-                _ => {
-                    panic!("Not a valid direction");
+                any => {
+                    return Err(GameError::InvalidDirection {
+                        direction: any.to_string(),
+                    })
```

Comment:
nit: unnecessary curly braces:
```suggestion
                any => return Err(GameError::InvalidDirection {
                    direction: any.to_string(),
                },
```